app/controllers/concerns/idv_step_concern.rb

```diff
@@ -108,6 +108,8 @@ def flow_policy
   end
 
   def confirm_step_allowed
```

Comment:
Maybe it worth considering to inject `sp_ession/decoreate_sp_session` into step `preconditions`, since we may need more and more `context` information to make decisions.